Sources/Mixpanel.swift

```diff
@@ -210,52 +210,57 @@ class MixpanelManager {
     static let sharedInstance = MixpanelManager()
     private var instances: [String: MixpanelInstance]
     private var mainInstance: MixpanelInstance?
-    private let readWriteLock: ReadWriteLock
-    private let instanceQueue: DispatchQueue
```

Comment:
It seems the separated lock and queue were not working well. 
there were some cases reading `mainInstance` inside readWriteLock's scope without mutex